Library/Homebrew/livecheck/strategy/git.rb

```diff
@@ -86,6 +86,21 @@ def self.find_versions(url, regex = nil)
 
           tags_only_debian = tags_data[:tags].all? { |tag| tag.start_with?("debian/") }
 
+          if block
+            case (value = block.call(tags_data[:tags], regex))
```

Comment:
@samford Might be nice to have some tests for these?

Reply:
I would love to test more code in the livecheck strategies but we're currently limited by what we can exercise without making a network call (or stubbing).

The only way to test this at the moment would be to call `Git#find_versions`, which calls `Git#tag_info`, which makes a network call. We tried this before and it worked locally but inexplicably failed on CI. Adding this type of test would make CI unreliable and frustrating, so we simply avoid it.

However, I have two PRs planned that will improve this situation:

* Add `provided_content` support to all the strategies (not just `PageMatch`). This continues to lay groundwork for caching response data in the future but also allows us to test the `#find_versions` methods by passing in static content (like we're doing in [`page_match_spec.rb`](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/livecheck/strategy/page_match_spec.rb)).
* Refactor the strategies to split the code that generates the `page_url` and default `regex` into a separate method (outside of `#find_versions`). This is needed to be able to implement smarter page caching (i.e., only storing page data if we'll need it again in a given run) but it also allows us to test this part of `#find_versions`.

Long story short, I can't add a test for this here but I'll be adding a test for it in a later PR. The aforementioned PRs are pretty simple, so I'm planning to have them up relatively soon (after I've wrapped up some others).

Reply:
> The only way to test this at the moment would be to call `Git#find_versions`, which calls `Git#tag_info`, which makes a network call. We tried this before and it worked locally but inexplicably failed on CI. Adding this type of test would make CI unreliable and frustrating, so we simply avoid it.

I think in this case it's worth stubbing, them?

Reply:
> I think in this case it's worth stubbing

It makes sense to me but I've avoided it up to this point since you [had qualms with stubbing](https://github.com/Homebrew/brew/pull/8180#issuecomment-669825147) when I initially suggested it. I agree that it's not ideal but I think it's at least better than the current situation, where we're not testing most livecheck code that involves network requests.

I'll work on this after the aforementioned PRs, as those will expand testing and we may be in a better position to decide how we should handle stubbing, where it makes sense, etc. However, if you have a particular approach in mind, let me know.

Reply:
Reliable integration tests that hit the network > stubbed network requests > unreliable integration tests that hit the network > no tests 😁